fretboard/src/main/java/mozilla/components/service/fretboard/ExperimentEvaluator.kt

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.fretboard
+
+import android.content.Context
+import android.os.Build
+import android.text.TextUtils
+import java.util.Locale
+import java.util.zip.CRC32
+
+internal class ExperimentEvaluator {
+    fun evaluate(
+        context: Context,
+        experimentDescriptor: ExperimentDescriptor,
+        experiments: List<Experiment>,
+        userBucket: Int = getUserBucket(context)
+    ): Boolean {
+        val experiment = experiments.firstOrNull { it.id == experimentDescriptor.id } ?: return false
+        return isInBucket(userBucket, experiment) && matches(context, experiment)
+    }
+
+    private fun matches(context: Context, experiment: Experiment): Boolean {
+        if (experiment.match != null) {
+            val region = Locale.getDefault().isO3Country
```

Comment:
Region is something different and not using the locale: On some devices we do a geo IP lookup once after app install to determine the location of the user in order to do thing like define the list of default search engines. We use geo IP for that because it is usually more accurate than the locale (e.g. I'm in Germany but using en-US). This is what we use for matching the region (while country/language use the locale data).

What we could do here is (either in this PR or just open a follow-up issue and ignore region here): Pass an optional interface to Fretboard (e.g. RegionProvider) that returns a region and its up to the app where it gets it from. It needs to be optional because some apps (like Focus) do not have any region information. And here we only would try to match the region if we have a provider.

Reply:
I'm going to update the PR with it then